universal-application-tool-0.0.1/app/assets/javascripts/admin_validation.ts

```diff
@@ -0,0 +1,65 @@
+/** The validation controller provides basic client-side validation of admin form fields. */
+class AdminValidationController {
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_CREATE = '#question-settings input[name="newOptions[]"]';
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_EDIT = '#question-settings input[name="options[]"]';
+  static readonly MULTI_OPTION_QUESTION_OPTION_CLASS = ".cf-multi-option-input";
+
+  constructor() {
+    // Attach listener to admin program edit form.
+    const adminProgramEditForm = document.getElementById("question-form");
+    if (adminProgramEditForm) {
+      adminProgramEditForm.addEventListener("submit", (event) => this.attemptSubmit(event));
+    }
+  }
+
+  private attemptSubmit(event: Event) {
+    this.validateForm();
+    if (!this.validateForm()) {

```

Comment:
suggestion (blocking): no need to call `validateForm` twice

Reply:
Fixed.